Source/swiftlint/Commands/LintCommand.swift

```diff
@@ -36,6 +36,15 @@ struct LintCommand: CommandType {
         let reporter = reporterFromString(
             options.reporter.isEmpty ? configuration.reporter : options.reporter
         )
+
+        // Validate that current version is newer than or equals to required version
+        if let required = configuration.requiredVersion {
+            if Version.current < required {
+                return .Failure(.UsageError(description: "configuration error: Your version \(Version.current) is" +
```

Comment:
I'm not sure if this should be a failure or a warning while proceeding to lint. Thoughts @realm/swiftlint ? I wouldn't want to necessarily break people's CI builds if someone updates the swiftlint version or somehow the CI provider's swiftlint version changes, but all the linting still passes.